.buildkite/plank-pipeline.yml

```diff
@@ -0,0 +1,39 @@
+steps:
+  - name: ":docker: Build base image with sources"
+    plugins:
+      docker-compose#v1.5.0:
+        build: app
+        image-repository: buildkite-registry.pinadmin.com/registry
+        volumes:
+          - ".:/app/mnt/buildkite-builds/volumes"
+        config: docker-compose.yml
+    agents:
+      queue: pinboard
+  - wait: ~
+  - name: "Build :iphone: :airplane:"
+    command: "swift build"
+    timeout_in_minutes: 30
+    agents:
+      queue: pinboard
+    plugins:
+      docker-compose#v1.5.0:
+        run: app
+        config: docker-compose.yml
+  - name: "Test :iphone: :airplane:"
+    command: "swift test"
+    timeout_in_minutes: 30
+    agents:
+      queue: pinboard
+    plugins:
+      docker-compose#v1.5.0:
+        run: app
+        config: docker-compose.yml
+  - name: "Integration Test :iphone: :airplane:"
+    command: "swift build && ./Utility/integration-test.sh"
```

Comment:
You probably want to move `swift build` either into the Docker container (if you're adding all files), or run this before integration-test.sh.

You might also create a single shell script that wraps both of these commands, but I think you can only run a single thing in the command here.

Reply:
I just saw that you were copying everything into the docker container with `COPY . /plank`. Because we're copying files this will invalidate the docker cache for basically every PR, which is fine, but might mean that we want to also run `swift build` inside of the docker container. This would also let you use the build for other steps in the pipeline if needed in parallel to integration tests.